package-build.el

```diff
@@ -1080,9 +1094,10 @@ and a cl struct in Emacs HEAD.  This wrapper normalises the results."
 
     (package-build--message "\n;;; %s\n" file-name)
     (let* ((version (package-version-join
-                     (version-to-list
+                     (package-build--version-to-list
```

Comment:
Right now I've got the checkouts (at least git and hg) returning the original version-string.  `package-build--find-version-newest` uses the version list internally to figure out which tag is newest, but then returns the original tag as a string so git and hg can actually check out that release.  But if SVN, darcs, etc., return the actual list, then I need to change this.

Should `package-build--find-version-newest` return something like `'((0 1) . "v0.1")`?  Then git/hg have access to the tag string for checkouts, but the checkout function itself could just return the `car`.